documentation/docs/30-advanced/70-packaging.md

```diff
@@ -126,6 +126,31 @@ This is a legacy field that enabled tooling to recognise Svelte component librar
 }
 ```
 
+### sideEffects
+
+The `sideEffects` option is used by bundlers to determine if a module may contain code that has side effects. A module is considered to have "side effects" if it makes changes that are observable from other scripts outside the module when it's imported; such as modifying global variables or the prototype of built-in JavaScript objects. Because a side effect could potentially affect the behavior of other parts of the application, these files/modules will be included in the final bundle regardless of whether their exports are used in the application.
+
+By adding `"sideEffects": false` in the `package.json`, you're signaling to the bundler that your package doesn't include any modules with side effects. This information can help the bundler to be more aggressive in eliminating unused exports from the final bundle, a process known as tree-shaking. This results in smaller and more efficient bundles. 
+
+> In the case of a Svelte component library, this prevents CSS from components that are not being used from being included in the final build.
```

Comment:
Can you confirm that setting sideEffects false for Svelte files doesn't have any unintended consequences, eg. even imported components not getting their CSS included?

Reply:
I'm definitely not an expert on this, but I have tested with [my library](https://github.com/rossrobino/drab) and everything is working as expected. I looked at a few other libraries with styles and have found they have this specified as well.

- [svelte ui](https://github.com/svelteuidev/svelteui/blob/main/packages/svelteui-core/package.json)
- [attractions](https://github.com/illright/attractions/blob/main/attractions/package.json)
- [carbon components](https://github.com/carbon-design-system/carbon-components-svelte/blob/master/package.json)

This is the only way I found to prevent the issue of all of the CSS being bundled, but open to suggestions if there is a better way to prevent this.

Reply:
I'm not sure if this is correct.

Carbon Svelte ships pre-compiled CSS stylesheets that the consumer imports in their app. The [`sideEffects: ['css/*.css']`](https://github.com/carbon-design-system/carbon-components-svelte/blob/b00ccc1d92e51cdd416739b29b150f19c3282301/package.json#L10-L12) field is necessary **explicitly for** bundlers that tree shake when building for production.

 One example is webpack. From the [webpack docs](https://webpack.js.org/guides/tree-shaking/):

> Note that any imported file is subject to tree shaking. This means if you use something like css-loader in your project and import a CSS file, it needs to be added to the side effect list so it will not be unintentionally dropped in production mode:

In the case of Carbon, if the user has the following:

```ts
import "carbon-components-svelte/css/all.css";
```

When building for production, Webpack will exclude that file if `sideEffects` does not explicitly include those CSS files. As a result, the stylesheet will be unexpectedly tree-shaken.

---

On the contrary, if the CSS stylesheet is imported in the [`Button`](https://github.com/carbon-design-system/carbon-components-svelte/blob/master/src/Button/Button.svelte) component (that is then used by the consumer), it will not be excluded as it becomes a part of the dependency tree.

Reply:
It sounds like perhaps our recommendation should be to set `"sideEffects": ["**/*.css"]`? I think that would lead to the desired outcome (https://github.com/webpack-contrib/mini-css-extract-plugin/issues/118#issuecomment-399738756). Most users will simply be distributing `.svelte` files in which case I suppose there would be no `.css` to match and so generally `"sideEffects": false` would be equivalent, but `"sideEffects": ["**/*.css"]` could be a little safer and avoid users running into trouble as soon as they add `.css`. I would love for others to confirm or test this as I don't have a ton of expertise in this area

Reply:
This works, I published my test repo and added this example in case this helps: https://github.com/rossrobino/side-effects-test

You can see the build outputs with sideEffects set to `false`, `true`, and `["**/*.css"]`. 

It can be tested by editing `node_modules/drab/package.json`. 

The imported `Breakpoint` component in `+page.svelte` has no CSS.

Reply:
@dominikg has reported:

>you can basically put anything "sideEffects":["bla"] has the same result as "sideEffects":["**/*.css"], so it looks like you have to declare something as having a side effect for the virtual modules getting tagged as free

Reply:
FWIW, my feeling is that for the majority of component libs shipped/built with SvelteKit, `sideEffects` is not needed **if** the components themselves use scoped styles (or import stylesheets).

I would say that Carbon is the exception (shipping individual stylesheets that the consumer must pick from).

Reply:
components using scoped styles exported through a barrel index.js files are the ones having the problem. Their css is added even if you never use them as soon as you import a single one from a barrel, all the css gets included

Reply:
Hmm. I just tried setting `sideEffects: ["**/*.css"]` in the `site-kit` package and the built CSS for kit.svelte.dev in this repo seemed to have gotten much larger as a result. I'm not sure which CSS exactly changed. It will take some more investigation

Reply:
Okay, this is _weird_. `sideEffects: ["**/*.css"]` and `sideEffects: false` in `site-kit` both massively bloat the CSS for kit.svelte.dev. E.g. you can see the `:root` CSS ends up in `0.css` in addition to one of the `_layout` CSS files. Omitting the `sideEffects` field or putting `sideEffects: []` result in smaller CSS output. I don't know how that makes any sense because what's the difference between `sideEffects: false` and `sideEffects: []`?

Reply:
That is weird. For what it's worth I'm seeing the same in my [tester](https://github.com/rossrobino/side-effects-test). When I set `sideEffects: []`, output is the same as `sideEffects: true`.

Reply:
https://github.com/vitejs/vite/pull/16152 was merged to fix this. I'm guessing that will be a part of Vite 5.2. We can revisit this docs PR once that's out